### PR TITLE
Prac 09 feedback

### PR DESCRIPTION
**Pull request by BrockAllen-au**
Hi @zachmorse1, please review my week 9 prac. Any and all feedback welcomed :-)

### CODE REVIEW

File: prac_09/taxi_test.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from taxi import Taxi
+
+my_taxi = Taxi("Prius 1", 100, 1.23)
+my_taxi.drive(40)
+my_taxi.start_fare()
+my_taxi.drive(100)
+print(my_taxi)
```
Comment (zachmorse1) [inline prac_09/taxi_test.py:7]: looks good to me

File: prac_09/band.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        return f"{self.name} ({band})"
+
+    def add(self, musician):
+        """Adds a musician to the band."""
+        self.musicians.append(musician)
+
+    def play(self):
+        """Get all band members to play their instruments."""
+        for person in self.musicians:
+            print(person.play())
```
Comment (zachmorse1) [inline prac_09/band.py:24]: looks good to me

File: prac_09/guitar.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+"""
+CP1404/CP5632 Practical - Suggested Solution
+Guitar class
+"""
+# Note: It would be better to get the current year from the system clock
+# and you might like to do that: lookup the "datetime" module
+CURRENT_YEAR = 2022
+VINTAGE_AGE = 50
```
Comment (zachmorse1) [inline prac_09/guitar.py:8]: good use of constants

File: prac_09/guitar.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        """Get the age of a guitar based on the CURRENT_YEAR."""
+        return CURRENT_YEAR - self.year
+
+    def is_vintage(self):
+        """Determine if a Guitar is considered vintage or not based on age."""
+        return self.get_age() >= VINTAGE_AGE
+
+    def __lt__(self, other):
+        """Less than, used for sorting Guitars - by year released."""
+        return self.year < other.year
```
Comment (zachmorse1) [inline prac_09/guitar.py:38]: Good use of docstrings

File: prac_09/silver_service_taxi.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        self.price_per_km = Taxi.price_per_km * fanciness
+        self.flagfall = 4.5
+
+    def __str__(self):
+        """String representation of Silver Service Taxi and fare."""
+        return f"{super().__str__()} plus flagfall of ${self.flagfall:.2f}"
+
+    def get_fare(self):
+        """Return the price for the taxi trip."""
+        return super().get_fare() + self.flagfall
```
Comment (zachmorse1) [inline prac_09/silver_service_taxi.py:23]: Looks good!

File: prac_09/unreliable_car.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    def __init__(self, name, fuel, reliability):
+        super().__init__(name, fuel)
+        self.reliability = reliability
+
+    def __str__(self):
+        return f"{super().__str__()}, reliability: {self.reliability}"
+
+    def drive(self, distance):
+        """Drive only if distance provided is less than the reliability provided."""
+        reliability_test = random.randint(0, 100)
```
Comment (zachmorse1) [inline prac_09/unreliable_car.py:21]: I would suggest using a better variable name. Such as "random_number"

File: prac_09/unreliable_car.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        if reliability_test > self.reliability:
+            distance = 0
+        elif distance > self.fuel:
```
Comment (zachmorse1) [inline prac_09/unreliable_car.py:24]: You should only need an if, no else pattern here. You can also simplify it by just returning the distance method through the super function.

File: prac_09/unreliable_car_test.py [hunk 8]
```diff
@@ -0,0 +1,10 @@
+"""Test file for testing the unreliable_car class"""
+
+from unreliable_car import UnreliableCar
+
+old_trusty = UnreliableCar("Old Trusty", 100, 50)
+print(old_trusty)
+
+old_trusty.drive(60)
+
+print(old_trusty)
```
Comment (zachmorse1) [inline prac_09/unreliable_car_test.py:10]: Good code! I could hardly fault it

Comment (BrockAllen-au) [reply prac_09/unreliable_car_test.py:10]: Thanks! 👍

**Pull request merged**